### PR TITLE
Fix a bug in the implementation that causes all Spans with RECORD_EVENTS options to also be exported.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/census-instrumentation/opencensus-java/issues/459

### CODE REVIEW

File: impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java [hunk 1]
```diff
@@ -57,17 +57,19 @@
   private final SpanContext notSampledSpanContext =
       SpanContext.create(
           TraceId.generateRandomId(random), SpanId.generateRandomId(random), TraceOptions.DEFAULT);
-  private final SpanExporterImpl sampledSpansServiceExporter = SpanExporterImpl.create(4, 1000);
-  private final StartEndHandler startEndHandler =
-      new StartEndHandlerImpl(sampledSpansServiceExporter, null, null, new SimpleEventQueue());
+  private final SpanExporterImpl spanExporter = SpanExporterImpl.create(4, 1000);
+  private final RunningSpanStoreImpl runningSpanStore = new RunningSpanStoreImpl();
+  private StartEndHandler startEndHandler =
+      new StartEndHandlerImpl(spanExporter, runningSpanStore, null, new SimpleEventQueue());
```
Comment (sebright) [inline impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java]: Shouldn't startEndHandler remain final?

File: impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java [hunk 2]
```diff
@@ -158,7 +160,7 @@ public void serviceHandlerThrowsException() {
   @Test
   public void exportSpansToMultipleServices() {
     FakeServiceHandler serviceHandler2 = new FakeServiceHandler();
```
Comment (g-easy) [inline impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java]: Could you please add a regression test that would have caught the original problem?

File: impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java (hunk 1 above)
Comment (bogdandrutu) [inline impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java]: Done.

File: impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java (hunk 2 above)
Comment (bogdandrutu) [inline impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java]: exportNotSampledSpans() will fail. I will add more explicit tests in a followup PR.

Comment (g-easy) [inline impl_core/src/test/java/io/opencensus/trace/export/SpanExporterImplTest.java]: Ok. :)

**Pull request merged**